realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -535,6 +534,45 @@ public RealmSchema getSchema() {
     }
 
     /**
+     * Removes all objects from this Realm.
+     *
+     * @throws IllegalStateException if the corresponding Realm is closed or in an incorrect thread.
+     */
+    void removeAll() {
+        checkIfValid();
+        for (RealmObjectSchema objectSchema : schema.getAll()) {
+            schema.getTable(objectSchema.getClassName()).clear();
+        }
+    }
+
+    /**
+     * Remove the given object from Realm which it is associated with.
+     */
+    static void remove(RealmObject realmObject) {
+        realmObject.removeFromRealm();
+    }
+
+    /**
+     * Remove the given objects from Realm which they are associated with.
+     */
+    static <E extends RealmObject> void remove(Iterable<E> realmObjects) {
+        if (realmObjects instanceof RealmResults)  {
+            ((RealmResults) realmObjects).clear();
+        } else if (realmObjects instanceof RealmList && ((RealmList) realmObjects).isValid()) {
+            ((RealmList) realmObjects).removeAllFromRealm();
+        } else {
+            for (RealmObject object : realmObjects) {
```

Comment:
If `realmObjects` is a `RealmList` and `realmObjects.isValid()` return `false`, you'll get into this block, right?